### PR TITLE
liquibase initial schema

### LINKED ISSUE #33: liquibase initial schema

### PR DESCRIPTION
**Pull request by h1alexbel**
@l3r8yJ take a look, please
closes #33

---

## PR-Codex overview
This PR focuses on adding an initial database schema and configuring Liquibase and R2DBC. 

### Detailed summary
- Added `001-initial-schema.xml` file with initial database schema.
- Added Liquibase and R2DBC dependencies in `pom.xml`.
- Configured Liquibase and R2DBC properties in `application.yaml`.
- Updated test class `HubApplicationEntryTest`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by h1alexbel** [comment]
@l3r8yJ what do you think about this schema?

coverage can be bumped after we create some objects in repositories

File: src/main/resources/db/changelog/2023/001-initial-schema.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    (
+        id BIGINT PRIMARY KEY,
+        created TIMESTAMPTZ DEFAULT now() NOT NULL
+    );
+    </sql>
+    <sql>
+    CREATE TABLE IF NOT EXISTS token
+    (
+        id BIGSERIAL PRIMARY KEY,
+        tkn CHARACTER VARYING(256) NOT NULL,
```
Comment (l3r8yJ) [inline src/main/resources/db/changelog/2023/001-initial-schema.xml:38]: maybe
```suggestion
        token CHARACTER VARYING(256) NOT NULL,
```

File: src/main/resources/db/changelog/2023/001-initial-schema.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-2.0.xsd" logicalFilePath="001-initial-schema.xml">
+  <changeSet id="001" author="h1alexbel">
+    <sql>
+    CREATE TABLE IF NOT EXISTS chat
+    (
+        id BIGINT PRIMARY KEY,
```
Comment (l3r8yJ) [inline src/main/resources/db/changelog/2023/001-initial-schema.xml:30]: i think it makes sense, wdyt
```suggestion
        id BIGINT UNIQUE PRIMARY KEY,
```

File: src/main/resources/db/changelog/2023/001-initial-schema.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        created TIMESTAMPTZ DEFAULT now() NOT NULL
+    );
+    </sql>
+    <sql>
+    CREATE TABLE IF NOT EXISTS token
+    (
+        id BIGSERIAL PRIMARY KEY,
```
Comment (l3r8yJ) [inline src/main/resources/db/changelog/2023/001-initial-schema.xml:37]: same here

File: pom.xml [hunk 4]
```diff
@@ -145,6 +166,16 @@ SOFTWARE.
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
         <version>${maven-surefire-plugin.version}</version>
+        <configuration>
```
Comment (l3r8yJ) [inline pom.xml:169]: I'm still wondering which approach is better, using `yml` configurations with profiles or this, wdyt?

**Review by l3r8yJ — CHANGES_REQUESTED** [review]
@h1alexbel take a look at my comments below, please

File: src/main/resources/db/changelog/2023/001-initial-schema.xml (hunk 2 above)
Comment (h1alexbel) [reply src/main/resources/db/changelog/2023/001-initial-schema.xml:30]: @l3r8yJ by default, PRIMARY KEY means UNIQUE + NOT NULL

read [this](https://stackoverflow.com/a/20404899/19147117)

File: pom.xml (hunk 4 above)
Comment (h1alexbel) [reply pom.xml:169]: @l3r8yJ we can try `yml`, but in bot, we already using XML config properties

File: src/main/resources/db/changelog/2023/001-initial-schema.xml (hunk 1 above)
Comment (h1alexbel) [reply src/main/resources/db/changelog/2023/001-initial-schema.xml]: accepted

File: pom.xml (hunk 4 above)
Comment (l3r8yJ) [reply pom.xml:169]: @h1alexbel yes, I think to use `yml` in the bot as well

Comment (h1alexbel) [reply pom.xml:169]: @l3r8yJ can we create a ticket for now? Since I don't know how to do it yml now

Comment (l3r8yJ) [reply pom.xml:169]: @h1alexbel yeah, go ahead

Comment (h1alexbel) [reply pom.xml:169]: @l3r8yJ created, https://github.com/Blamer-io/hub/issues/37 and https://github.com/Blamer-io/bot/issues/59

**Comment by l3r8yJ** [comment]
@rultor merge

**Comment by l3r8yJ** [comment]
@h1alexbel fix the codecov action so it does not fails

**Comment by h1alexbel** [comment]
@l3r8yJ I've tested `fail_on_ci: false`, but it still fail the build

**Comment by l3r8yJ** [comment]
so, manual merge(

**Pull request merged**